src/Promitor.Core/Metrics/MeasuredMetric.cs

```diff
@@ -15,12 +16,12 @@ public class MeasuredMetric
         /// <summary>
         ///     Name of dimension for a metric
         /// </summary>
-        public string DimensionName { get; set; }
+        public List<string> DimensionNames { get; set; }
 
         /// <summary>
         ///     Name of dimension for a metric
         /// </summary>
-        public string DimensionValue { get; }
+        public List<string> DimensionValues { get; }
```

Comment:
Unless C# Lists do not guarantee keeping their order, it should always be correct, but we wondered about whether that might simply be easier to follow, too.
Since it also helps with the issues with the two for loops you commented on, I refactored this now.